src/MBrace.Azure/Management.fs

```diff
@@ -338,177 +479,331 @@ module private Details =
+    /// <param name="logger">System logger used by the manager instance. Defaults to no logging.</param>
+    /// <param name="logLevel">Log level used by the manager instance. Defaults to Info.</param>
+    static member FromPublishSettingsFile(publishSettingsFile : string, defaultRegion : Region, ?defaultSubscriptionId : string, ?logger : ISystemLogger, ?logLevel : LogLevel) =
+        let pubSettings = PublishSettings.ParseFile publishSettingsFile
+        DeploymentManager.Create(pubSettings.Subscriptions, defaultRegion, ?defaultSubscriptionId = defaultSubscriptionId, ?logger = logger, ?logLevel = logLevel)
+
+
+    //
+    //  Static APIs
+    //
+
+    /// <summary>
+    ///     Starts deployment of MBrace cloud service with supplied parameters.
+    /// </summary>
+    /// <param name="publishSettingsFile">Path to local PublishSettings file.</param>
+    /// <param name="region">Azure region for deployments.</param>
+    /// <param name="vmSize">VM size used for deployment.</param>
+    /// <param name="vmCount">VM instance count.</param>
+    /// <param name="serviceName">Service name identifier. Defaults to auto-generated name.</param>
+    /// <param name="subscriptionId">Subscription identifier for service deployment. Defaults to manager instance default subscription.</param>
+    /// <param name="region">Region for service deployment. Defaults to manager instance default region.</param>
+    /// <param name="mbraceVersion">MBrace version string used for .cspkg resolution. Defaults to current version.</param>
+    /// <param name="storageAccount">Storage account name or connection string used by MBrace service. Defaults to self-allocated storage account.</param>
+    /// <param name="serviceBusAccount">Service bus account name or connection string used by MBrace service. Defaults to self-allocation service bus account.</param>
+    /// <param name="cloudServicePackage">Path or Uri to MBrace cloud service package to be deployed to Service. Defaults to .cspkg resolved from github.</param>
+    /// <param name="serviceLabel">User-supplied service label.</param>
+    static member BeginDeploy(pubSettingsFile : string, region : Region, vmSize : string, vmCount : int, ?serviceName : string, ?subscriptionId : string, 
+                                ?mbraceVersion : string, ?storageAccount : string, ?serviceBusAccount : string, ?cloudServicePackage : string, ?serviceLabel : string) =
+        let manager = DeploymentManager.FromPublishSettingsFile(pubSettingsFile, defaultRegion = region, ?defaultSubscriptionId = subscriptionId, logger = new ConsoleLogger(true))
+        manager.BeginDeploy(vmSize, vmCount, ?serviceName = serviceName, ?subscriptionId = subscriptionId, ?mbraceVersion = mbraceVersion, 
+                                ?storageAccount = storageAccount, ?serviceBusAccount = serviceBusAccount, ?cloudServicePackage = cloudServicePackage, ?serviceLabel = serviceLabel)
+
+
```

Comment:
There should be, but the `Deployments.CreateAsync()`  method seems to be failing with http exceptions at the moment. In this case I would favor replace `Deploy`/`BeginDeploy` with a deploy method that returns a progress token, much like CloudProcess in the base API.